microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/resources/com/redhat/microprofile/jdt/internal/metrics/properties/metadata.json

```diff
@@ -0,0 +1,18 @@
+[
+	{
+		"type": "java.util.Optional<java.lang.String>",
+		"required": true,
```

Comment:
are you sure ?

Reply:
> prefer using java.lang.String instead of using java.util.Optionnal<java.lang.String> (if it is the case)

For `mp.metrics.tags` and `mp.metrics.appName`, I think it should stay as java.util.Optional<java.lang.String> because of this code:
https://github.com/eclipse/microprofile-metrics/blob/da7045a380be7175e87da8585f053fb4593723e3/api/src/main/java/org/eclipse/microprofile/metrics/MetricID.java#L123

and this code:
https://github.com/eclipse/microprofile-metrics/blob/da7045a380be7175e87da8585f053fb4593723e3/api/src/main/java/org/eclipse/microprofile/metrics/MetricID.java#L127

But for the required, it should be false, right? Since the JSON was from the server trace, I will investigate why required is being set to true.

Reply:
The code is `config.getOptionalValue(GLOBAL_TAGS_VARIABLE, String.class);` the second parameter is a String, not the Optional, I suppose it's the call of config.getOptionalValue which transform the String to Optional<String>. So I think it's just a String.

> But for the required, it should be false, right? Since the JSON was from the server trace, I will investigate why required is being set to true.

You are right there is a bug. In PropertiesCollector we have
`property.setRequired(defaultValue == null);`

It should be:

`property.setRequired(defaultValue != null);`

Perhaps it should be fixed in an another PR, because I fear tests will not good. We need to regenereate all-quarkus-properties.json too I think. Please create an issue for that.

Reply:
To confirm, the type should be `java.lang.String` and required should stay `true`?

Reply:
> To confirm, the type should be java.lang.String 

yes

> and required should stay true

no, please change that for this usecase. For the automtaic update of required of defaultValue we will fix that in an another PR.

Reply:
Fixed